### PR TITLE
[Newton] RLM-1120 Use dist-upgrade for loose/no artifacts

### PR DESCRIPTION
**Pull request by antonym**
Ensures disto is fully updating including kernel
to surface any dependency issues.

Cherry-picked from https://github.com/rcbops/rpc-openstack/pull/2778

Issue: [RLM-1120](https://rpc-openstack.atlassian.net/browse/RLM-1120)

### CODE REVIEW

File: gating/pre_merge_test/run_deploy.sh [hunk 1]
```diff
@@ -48,7 +48,7 @@ if [[ ${RE_JOB_IMAGE} =~ no_artifacts$ ]]; then
   # Upgrade to the absolute latest
   # available packages.
   apt-get update
-  DEBIAN_FRONTEND=noninteractive apt-get -y upgrade
+  DEBIAN_FRONTEND=noninteractive apt-get -y dist-upgrade
```
Comment (odyssey4me) [inline gating/pre_merge_test/run_deploy.sh]: Doing this upgrade may upgrade the kernel packages, but won't make them active. It requires a reboot to be active as far as I know. Is this what you want?

File: gating/pre_merge_test/run_deploy_mnaio.sh [hunk 2]
```diff
@@ -109,11 +109,11 @@ env | grep RE_ | sed 's/^/export /' > /opt/rpc-openstack/RE_ENV
 # check if we're using artifacts or not
 if [[ ${RE_JOB_IMAGE} =~ no_artifacts$ ]]; then
   echo "export RPC_APT_ARTIFACT_ENABLED=no" >> /opt/rpc-openstack/RE_ENV
-  ${MNAIO_SSH} "apt-get -qq update; DEBIAN_FRONTEND=noninteractive apt-get -y upgrade"
+  ${MNAIO_SSH} "apt-get -qq update; DEBIAN_FRONTEND=noninteractive apt-get -y dist-upgrade"
```
Comment (odyssey4me) [inline gating/pre_merge_test/run_deploy_mnaio.sh]: Given that this is just a host for the VM's, is this even necessary?

File: gating/pre_merge_test/run_deploy.sh (hunk 1 above)
Comment (antonym) [reply gating/pre_merge_test/run_deploy.sh]: Yeah, @cloudnull and I chatted and given there are some package dependencies on the kernel, this should still simulate the dependency tree without the reboot.

File: gating/pre_merge_test/run_deploy_mnaio.sh (hunk 2 above)
Comment (antonym) [reply gating/pre_merge_test/run_deploy_mnaio.sh]: It's actually running the command on the infra1 host which will be doing the deployment.

Comment (antonym) [reply gating/pre_merge_test/run_deploy_mnaio.sh]: Technically this will probably noop given the VMs are freshly provisioned.

**Pull request merged**